<?php	class PhxException extends Exception{		public function __construct($message, $code = 0) {			$this->save_error_log($message);			parent::__construct($message, $code);		}				private function save_error_log($message){			$logger = Logger::getInstance();			$logger->file = "exception_log_" . date('Y_m_d');			$logger->message("Exception caught: {$message} \n" . print_r($this, true));		}	}